control-plane/csi-driver/controller/src/controller.rs

```diff
@@ -272,31 +263,32 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
         //
         // The requisite array contains all nodes in the cluster irrespective
         // of what node was chosen for running the app.
-        let mut allowed_nodes: Vec<String> = Vec::new();
-        let mut preferred_nodes: Vec<String> = Vec::new();
+        let mut allowed_nodes: HashSet<String> = HashSet::new();
+        let mut preferred_nodes: HashSet<String> = HashSet::new();
         let mut inclusive_label_topology: HashMap<String, String> = HashMap::new();
+        let supported_keys = vec![OPENEBS_TOPOLOGY_KEY, K8S_HOSTNAME_KEY];
 
         inclusive_label_topology.insert(String::from(CREATED_BY_KEY), String::from(DSP_OPERATOR));
 
         if let Some(reqs) = args.accessibility_requirements {
             for r in reqs.requisite.iter() {
                 for (k, v) in r.segments.iter() {
-                    // We are not able to evaluate any other topology requirements than
-                    // the hostname req. Reject all others.
-                    if k != K8S_HOSTNAME {
-                        return Err(Status::invalid_argument(
-                            "Volume topology other than hostname not supported",
-                        ));
+                    // Reject all others than `supported_keys`
+                    if !supported_keys.iter().any(|supported| supported == k) {
+                        return Err(Status::invalid_argument(format!(
+                            "Volume topology keys other than {} and {} not supported",
+                            K8S_HOSTNAME_KEY, OPENEBS_TOPOLOGY_KEY
+                        )));
```

Comment:
Nit: A suggestion for readability.
```suggestion
                   // Reject all others than `supported_keys`
                    if supported_keys.contains(&k.as_str()) {
                        allowed_nodes.insert(v.to_string());
                    } else {
                        return Err(Status::invalid_argument(format!(
                            "Volume topology keys other than {} and {} not supported",
                            K8S_HOSTNAME_KEY, OPENEBS_TOPOLOGY_KEY
                        )));
                    }
```